packages/engine-server/src/config.ts

```diff
@@ -49,29 +53,53 @@ export class DConfig {
     return config;
   }
 
-  static getSiteIndex(sconfig: DendronSiteConfig) {
-    let { siteIndex, siteHierarchies } = sconfig;
+  static getSiteIndex(sconfig: DendronSiteConfig | DendronPublishingConfig) {
+    const { siteIndex, siteHierarchies } = sconfig;
     return siteIndex || siteHierarchies[0];
   }
 
   /**
    * fill in defaults
    */
-  static cleanSiteConfig(config: DendronSiteConfig): CleanDendronSiteConfig {
-    let out: DendronSiteConfig = _.defaults(config, {
-      copyAssets: true,
-      usePrettyRefs: true,
-      siteNotesDir: "notes",
-      siteFaviconPath: "favicon.ico",
-      gh_edit_link: true,
-      gh_edit_link_text: "Edit this page on GitHub",
-      gh_edit_branch: "main",
-      gh_root: "docs/",
-      gh_edit_view_mode: "edit",
-      writeStubs: true,
-      description: "Personal knowledge space",
-    });
-    let { siteRootDir, siteHierarchies, siteIndex, siteUrl } = out;
+  static cleanSiteConfig(
```

Comment:
might be cleaner to split this into 2 methods (cleanSiteConfig and cleanLegacySiteConfig) and only have an if condition checking for this at the caller side (the caller still needs to handle the configs differently anyways)